parlai/agents/transformer/modules.py

```diff
@@ -144,6 +144,9 @@ def encode_cand(self, words):
 
     def encode_context_memory(self, context_w, memories_w):
         # [batch, d]
+        if context_w is None:
```

Comment:
Why did this change here?

Reply:
oh nvm, I see below. So Transformer wasn't even using the caching before?

Reply:
Actually, I'm still confused. Nothing about context memory should ever be cached right?

Reply:
correct - transformer was not caching before. 

this is just saying if context is None, we want to return None. (useful for when we want to encode just the RHS)

Reply:
So I’m still a tad bit confused why then this guard ends up in the encode_context_memory method

Reply:
Anyway maybe you can just add a short comment here saying why context_w might be None and then it lgtm.

Reply:
"this is just saying if context is None, we want to return None. "
I am confused. If context is None, shouldn't we return the embedding of the candidates only?

Reply:
Ah, nevermind, wrong function